disttask/framework/storage/task_table.go

```diff
@@ -39,6 +39,8 @@ import (
 type SessionExecutor interface {
 	// WithNewSession executes the function with a new session.
 	WithNewSession(fn func(se sessionctx.Context) error) error
+	// WithNewTxn executes the fn in a new transaction.
+	WithNewTxn(ctx context.Context, fn func(se sessionctx.Context) error) error
```

Comment:
These naming is really misleading 😂 WithNewTxn makes me think it's an option for builder, and the parameter is the txn provided to builder